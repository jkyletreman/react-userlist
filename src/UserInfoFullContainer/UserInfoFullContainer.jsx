import React, { Component } from "react";
import UserInfoFull from "../UserInfoFullPresentation/UserInfoFull";

export default class UserInfoFullContainer extends Component {

  calculateUserBirthYear = () => {
    const currentTime = new Date();
    const year = currentTime.getFullYear();
    return year - this.props.selectedUser.age;
  };

  render() {
    return (
      <UserInfoFull
        selectedUser={this.props.selectedUser}
        birthYear={this.calculateUserBirthYear()}
      />
    );
  }
}
