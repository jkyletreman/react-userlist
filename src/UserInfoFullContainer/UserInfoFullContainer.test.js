import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import UserInfoFullContainer from './UserInfoFullContainer'
import ShallowRenderer from 'react-test-renderer/shallow';

it('renders user shallow correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<UserInfoFullContainer selectedUser={{age: 24}} />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
