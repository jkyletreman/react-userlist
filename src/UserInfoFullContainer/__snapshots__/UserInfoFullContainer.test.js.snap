// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user shallow correctly 1`] = `
<UserInfoFull
  birthYear={1994}
  selectedUser={
    Object {
      "age": 24,
    }
  }
/>
`;
